src/extensions/default/CodeFolding/foldhelpers/foldgutter.js

```diff
@@ -214,7 +211,7 @@ define(function (require, exports, module) {
             var lineChanges = changeObj.text.length - changeObj.removed.length;
             //update the lineFolds cache
             updateFoldsCache(cm, changeObj.from.line, lineChanges);
-            if (lineChanges !== 0) {
+            if (lineChanges !== 0 && changeObj.from.line + lineChanges >= 0) {
```

Comment:
I'm not sure I understand this part of the change. The only time `changeObj.from.line + lineChanges < 0` could happen is when a bunch of lines were deleted at the start of the file.  Do we actually want to skip running `updateFoldInfo()` in that case?

I'm wondering if what we really want is something more like `Math.max(0, changeObj.from.line + lineChanges)` on the line below this instead... and similar for the 2nd arg too.